canvasapi/quiz.py

```diff
@@ -240,26 +244,165 @@ def get_all_quiz_submissions(self, **kwargs):
+
+        :rtype: :class:`canvasapi.quiz.QuizSubmission`
+        """
+        response = self._requester.request(
+            'POST',
+            'courses/{}/quizzes/{}/submissions'.format(
+                self.course_id,
+                self.id
+            ),
+            _kwargs=combine_kwargs(**kwargs)
+        )
+
+        response_json = response.json()["quiz_submissions"][0]
+        response_json.update({'course_id': self.course_id})
+
+        return QuizSubmission(self._requester, response_json)
+
 
 @python_2_unicode_compatible
 class QuizSubmission(CanvasObject):
 
     def __str__(self):
         return "{}-{}".format(self.quiz_id, self.user_id)
 
+    def complete(self, **kwargs):
+        """
+        Complete the quiz submission by marking it as complete and grading it. When the quiz
+        submission has been marked as complete, no further modifications will be allowed.
+
+        :calls: `POST /api/v1/courses/:course_id/quizzes/:quiz_id/submissions/:id/complete \
+        <https://canvas.instructure.com/doc/api/quiz_submissions.html#method.quizzes/quiz_submissions_api.complete>`_
+
+        :rtype: :class:`canvasapi.quiz.QuizSubmission`
+        """
+        if 'attempt' in kwargs:
+            raise ValueError("Key `attempt` provided by Canvas, should not be set.")
+
+        if 'validation_token' in kwargs:
+            raise ValueError("Key `validation_token` provided by Canvas, should not be set.")
+
+        kwargs['attempt'] = self.attempt
+        kwargs['validation_token'] = self.validation_token
+
+        response = self._requester.request(
+            'POST',
+            'courses/{}/quizzes/{}/submissions/{}/complete'.format(
+                self.course_id,
+                self.quiz_id,
+                self.id
+            ),
+            _kwargs=combine_kwargs(**kwargs)
+        )
+
+        response_json = response.json()["quiz_submissions"][0]
+        return QuizSubmission(self._requester, response_json)
+
+    def get_times(self, **kwargs):
+        """
+        Get the current timing data for the quiz attempt, both the end_at timestamp and the
+        time_left parameter.
+
+        :calls: `GET /api/v1/courses/:course_id/quizzes/:quiz_id/submissions/:id/time \
+        <https://canvas.instructure.com/doc/api/quiz_submissions.html#method.quizzes/quiz_submissions_api.time>`_
+
+        :rtype: dict
+        """
+        if 'attempt' in kwargs:
+            raise ValueError("Key `attempt` provided by Canvas, should not be set.")
+
+        response = self._requester.request(
+            'GET',
+            'courses/{}/quizzes/{}/submissions/{}/time'.format(
+                self.course_id,
+                self.quiz_id,
+                self.id
+            ),
+            _kwargs=combine_kwargs(**kwargs)
+        )
+
+        response_json = response.json()
+        return response_json
```

Comment:
If you make no modifications to `response.json()`, there's no reason to save it out to a variable. Just return `response.json()` directly.